packages/react-composites/src/composites/ChatComposite/FileDownloadCards.tsx

```diff
@@ -50,18 +67,27 @@ export const FileDownloadCards = (props: FileDownloadCards): JSX.Element => {
             fileExtension={extension(file.name)}
             actionIcon={<DownloadIconTrampoline />}
             actionHandler={() => {
-              props.downloadHandler && props.downloadHandler(userId, file);
+              if (props.downloadHandler) {
+                // @conditional-compile-remove(file-sharing)
+                setSpinner(true);
+                props
+                  .downloadHandler(userId, file)
+                  .then((fileurl) => {
+                    // @conditional-compile-remove(file-sharing)
+                    setSpinner(false);
+                    window.open(fileurl.toString(), '_blank', 'noopener,noreferrer');
+                  })
+                  .catch((error) => {
+                    // @conditional-compile-remove(file-sharing)
+                    setSpinner(false);
```

Comment:
You can use a finally block to stop the spinner instead of multiple declarations :)

Reply:
Done